src/app/services/user.ts

```diff
@@ -19,6 +19,10 @@ export function isTutorOrAbove(user?: { readonly role?: UserRole; readonly logge
   return isDefined(user) && user.role !== "STUDENT" && (user.loggedIn ?? true);
 }
 
+export function isTeacher(user?: { readonly role?: UserRole; readonly loggedIn?: boolean } | null): boolean {
```

Comment:
I know that you're just emulating the other functions around this one, but this is an easy candidate for some tests. Also, the type of this `user` (`{ readonly role?: UserRole; readonly loggedIn?: boolean }`) seems to be used a lot in this file, so it could probably be extracted (not exported) to its own type.

Reply:
Added type, and 119 tests for the functions in `/services/user.ts`

Reply:
Thank you, that's great and a lot more than what I meant! I hope I didn't sound like I was asking you to test everything in this file, I only meant your new function. But thanks for improving the codebase ❤️